### PR TITLE
- [xorg-proto] component requirements are mandatory now

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **[xorg-proto]/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xorg-proto/all/conanfile.py [hunk 1]
```diff
@@ -112,6 +112,8 @@ def package_info(self):
             # FIXME: generated .pc files contain `Name: xorg-proto-Xproto`, it should be `Name: Xproto`
             self.cpp_info.components[filename].filenames["pkg_config"] = filename
             self.cpp_info.components[filename].libdirs = []
+            if hasattr(self, "settings_build"):
+                self.cpp_info.components[filename].requires = ["xorg-macros/1.19.3"]
```
Comment (SSE4) [inline recipes/xorg-proto/all/conanfile.py:116]:
```suggestion
                self.cpp_info.components[filename].requires = ["xorg-macros"]
```

File: recipes/xorg-proto/all/conanfile.py [hunk 2]
```diff
@@ -112,6 +112,8 @@ def package_info(self):
             # FIXME: generated .pc files contain `Name: xorg-proto-Xproto`, it should be `Name: Xproto`
             self.cpp_info.components[filename].filenames["pkg_config"] = filename
             self.cpp_info.components[filename].libdirs = []
+            if hasattr(self, "settings_build"):
+                self.cpp_info.components[filename].requires = ["xorg-macros"]
```
Comment (ericLemanissier) [inline recipes/xorg-proto/all/conanfile.py:116]:
```suggestion
                self.cpp_info.components[filename].requires = ["xorg-macros::xorg-macros"]
```

**Review by madebr — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Comment by SpaceIm** [comment]
I'm wondering if it's a good idea to have a conditional requirements() depending on 1 or 2 profiles.

**Comment by madebr** [comment]
Perhaps the conditional in `requirements` and `build_requirements` should be switched.

**Comment by jgsogo** [comment]
I'm ok with this PR, it works... but the comment by @SpaceIm is something to take into account, sure it is better to require unconditionally and have the same package-ID (any reason not to do it?), otherwise from the POV of an Apple M1 user (running 1 profile), the binary won't be available.

**Pull request merged**